lib/logproto/logproto-text-client.c

```diff
@@ -84,6 +96,14 @@ log_proto_text_client_process_input(LogProtoClient *s)
   return LPS_SUCCESS;
 }
 
+static LogProtoStatus
+_prohibit_in(LogProtoClient *s)
+{
+  g_assert_not_reached();
```

Comment:
With this assert `_prohibit_in()` essentially becomes "`_crash_on_input()`". I don't see, how this is a desirable behaviour, even if `process_in()` should not be called on an unidirectional client.

Reply:
this assert will signal a logic error in the IO state management; a unidirectional client should never read